### PR TITLE
chore/MM-17293 : Migrate 'components/invitation_modal/invitation_modal_guest_step' and associated tests to TypeScript

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Summary

Migrated `components/invitation_modal/invitation_modal_guest_step` folder files to ts

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/17293

#### Related Pull Requests

- Has server changes (please link here)
- Has mobile changes (please link here)

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by M-ZubairAhmed** [comment]
@DHaussermann can you please check for the bug we encountered earlier

**Review by devinbinnie — APPROVED** [review]
LGTM! Thanks @M-ZubairAhmed!

File: components/invitation_modal/invitation_modal_guest_step/index.ts [hunk 1]
```diff
@@ -32,4 +32,8 @@ function mapDispatchToProps(dispatch: Dispatch) {
     };
 }
 
-export default connect(mapStateToProps, mapDispatchToProps)(InvitationModalGuestsStep);
+const connector = connect(mapStateToProps, mapDispatchToProps);
+
+export type PropsFromRedux = ConnectedProps<typeof connector>;
```
Comment (hmhealey) [inline components/invitation_modal/invitation_modal_guest_step/index.ts:37]: Interesting. I didn't know about that helper before. I'm curious if it even types `props.actions.getSubscriptionStats` properly when it's a thunk action

**Review by hmhealey — APPROVED** [review]
Looks good to me!

File: components/invitation_modal/invitation_modal_guest_step/index.ts (hunk 1 above)
Comment (M-ZubairAhmed) [reply components/invitation_modal/invitation_modal_guest_step/index.ts:37]: yes it does can you check below
image

Comment (hmhealey) [reply components/invitation_modal/invitation_modal_guest_step/index.ts:37]: That's not quite correct then actually. `connect` should get rid of the `ActionFunc` return value

**Comment by hmhealey** [comment]
@DHaussermann Do you have a chance to look at this soon, or should we re-assign it?

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- Tested member invite
- Tested guest invite enabled /. disabled
- Tested all extra fields guest invite function properly
- Tested on browser and desktop
- No functional issues found
- No related JS errors in dev console
LGTM!
 
**note**: I did see a minor annoyance with invite functionality unrelated to this PR. Created  
https://mattermost.atlassian.net/browse/MM-38303 to address. 

@M-ZubairAhmed sorry for the delay in testing. Thank you for this PR. Much appreciated!

**Pull request merged**